### PR TITLE
Prepare the FabricBot config for migration to Policy Service

### PR DESCRIPTION
**Pull request by jeffhandley**
There are changes that need to be made to the FabricBot configuration before this repository is ready for migration to Policy Service. There is some lost functionality here, but it was minimized as much as possible.

1. Remove the auto-merge automation
    - This automation configuration was relying on some rich features of the FabricBot's own auto-merge functionality
    - That functionality was in place before GitHub's own auto-merge functionality
    - Because GitHub has built-in auto-merge capabilities, the Policy Service does not provide its own implementation
    - Policy Service does have automation that allows for a PR to have auto-merge enabled or disabled, but it can't rely on the types of conditions used here
    - The result is there is not a way to port this configuration forward, and a different approach is needed
    - The recommendation is to investigate Arcade-based auto-merge behaviors or implementing a GitHub Action
2. The 'Require PR approval before merging SDK change' cannot be ported
    - As noted in #71275, the functionality of requesting changes to a pull request through automation is not supported in the new service
    - There isn't a clear way to port this behavior
    - Since auto-merge automation is also being removed, this means the workflow for auto-merging some PRs but blocking others needs to be addressed differently after the migration to Policy Service
    - It's also recommended to look into Arcase-based auto-merge behaviors or implementing a GitHub Action
3. Explicitly add a filter for a pull request event and make 2 corrections to existing configuration
    - There was a task configured for `"eventType": "pull_request"` that when ported to Policy Service, this implicit filter will be lost
    - This updates the config to explicitly add that filter so behavior stays the same after the migration
    - **NOTE:** There were two tasks that were not previously working as seemingly intended, because the implicit filters were filtering out the expected issue/PR type. Those existing configurations are corrected here; they should start working after this PR is merged and continue working after the migration.
4. Remove references to project automation
5. Delete tasks that were marked as disabled

@jaredpar @arkalyanms @allisonchou Sorry for the loss of auto-merge functionality here; we explored ways to retain this, but we can't fill that gap for you.

/cc @wtgodbe @mkArtakMSFT @JohannesLampel

### CODE REVIEW

File: .github/fabricbot.json
Comment (jeffhandley) [inline .github/fabricbot.json:1]: I recommend reviewing using the Split/Side-by-Side view, as where tasks were deleted, the inline diff blends adjacent task configs together to make the changes look wrong.

File: .github/fabricbot.json [hunk 1]
```diff
@@ -271,10 +201,9 @@
             }
           ]
         },
-        "eventType": "issue",
+        "eventType": "pull_request",
```
Comment (jeffhandley) [inline .github/fabricbot.json:204]: This task was previously applying only to issues, but it was intended to apply to pull requests. This corrects the current configuration and also prepares it to port properly.

File: .github/fabricbot.json [hunk 2]
```diff
@@ -150,6 +119,10 @@
         "conditions": {
           "operator": "and",
           "operands": [
+            {
+              "name": "isPr",
+              "parameters": {}
+            },
```
Comment (jeffhandley) [inline .github/fabricbot.json:125]: When this task is ported over, the configuration for Policy Service will need to be hand-edited. The `merged` action is not supported on Policy Service, so instead the configuration will need to check for `closed` with the `isMerged` condition.

**Review by arkalyanms — APPROVED** [review]

File: .github/fabricbot.json [hunk 3]
```diff
@@ -199,58 +171,16 @@
     {
       "taskType": "trigger",
       "capabilityId": "IssueResponder",
-      "subCapability": "PullRequestResponder",
+      "subCapability": "IssuesOnlyResponder",
```
Comment (jjonescz) [inline .github/fabricbot.json:174]: Why IssuesOnlyResponder if this applies to PRs?

Comment (jeffhandley) [reply .github/fabricbot.json:174]: Interestingly, this task was previously marked as `IssuesOnlyResponder` and the diff display merely makes it look like this was edited. So it was wrong before too, and I'm unsure if the task was previously functioning properly.

I'll go ahead and fix it regardless. Good catch!

**Pull request merged**